toolbox/util/githubClient.go

```diff
@@ -164,10 +172,49 @@ func (g GithubClient) CloseFailedPullRequests(prTitlePrefix, repo, baseBranch st
 
 // GetHeadCommitSHA finds the SHA of the commit to which the HEAD of branch points
 func (g GithubClient) GetHeadCommitSHA(repo, branch string) (string, error) {
-	ref, _, err := g.client.Git.GetRef(
-		context.Background(), g.owner, repo, "refs/heads/"+branch)
+	return g.getReferenceSHA(repo, "refs/heads/"+branch)
+}
+
+// GetSHATime gets the time when sha is made
+func (g GithubClient) GetSHATime(repo, sha string) (*time.Time, error) {
+	commit, _, err := g.client.Git.GetCommit(
+		context.Background(), g.owner, repo, sha)
+	if err != nil {
+		return nil, err
+	}
+	return (*(*commit).Author).Date, nil
+}
+
+// GetTagPublishTime finds the time a tag is published
+func (g GithubClient) GetTagPublishTime(repo, tag string) (*time.Time, error) {
+	sha, err := g.getReferenceSHA(repo, "refs/tags/"+tag)
+	if err != nil {
+		return nil, err
+	}
+	tagObj, _, err := g.client.Git.GetTag(
+		context.Background(), g.owner, repo, sha)
+	if err != nil {
+		return nil, err
+	}
+	return (*(*tagObj).Tagger).Date, nil
+}
+
+// GetFileContent retrieves the file content from the hosted repo
+func (g GithubClient) GetFileContent(repo, branch, path string) (string, error) {
+	opt := github.RepositoryContentGetOptions{branch}
+	fileContent, _, _, err := g.client.Repositories.GetContents(
+		context.Background(), g.owner, repo, path, &opt)
+	if err != nil {
+		return "", err
+	}
+	return fileContent.GetContent()
```

Comment:
fileContent.GetContent() returns both a string and error so the trailing nil would be unnecessary.